### ISSUE TITLE
CA2256 false positive (?) on nested interface

### ISSUE THREAD

**Issue by alexrp** [dotnet/roslyn-analyzers#5964]
### Analyzer

**Diagnostic ID**: [CA2256](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2256): `All members declared in parent interfaces must have an implementation in a DynamicInterfaceCastableImplementation-attributed interface`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 6.0.201](https://dotnet.microsoft.com/download/dotnet/6.0)

### Describe the bug

CA2256 is (I believe) firing incorrectly on nested interfaces attributed as `[DynamicInterfaceCastableImplementation]`.

### Steps To Reproduce

```csharp
public interface ICLRRuntimeLocator
{
    [DynamicInterfaceCastableImplementation]
    internal interface ICLRRuntimeLocatorWrapper : ICLRRuntimeLocator
    {
        ulong ICLRRuntimeLocator.GetRuntimeBase()
        {
            return 0;
        }
    }

    public static readonly Guid IID_ICLRRuntimeLocator = new("b760bf44-9377-4597-8be7-58083bdc5146");

    ulong GetRuntimeBase();
}
```

### Expected behavior

No warning.

### Actual behavior

Warning fires on `ICLRRuntimeLocatorWrapper`.

## Additional context

Moving `ICLRRuntimeLocatorWrapper` out of `ICLRRuntimeLocator` so it is no longer nested gets rid of the warning.

**Comment by Youssef1313** [comment]
https://github.com/dotnet/roslyn-analyzers/blob/e898a9d806d66adf687e2e3eb3d0180ca8a2167a/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/DynamicInterfaceCastableImplementation.cs#L102

I think this should be `if (!member.IsStatic && member.Kind != SymbolKind.NamedType`

@alexrp Would you like to open a PR and fix this?

**Comment by Youssef1313** [comment]
Is `member.IsStatic` check is actually correct given the abstract statics in interfaces feature? 

cc @jkoritzinsky

**Comment by jkoritzinsky** [comment]
Yeah we should probably remove the IsStatic check now that statics can be missing implementations in the base interface.

**Comment by Youssef1313** [comment]
Probably it should be replaced with `member.IsVirtual || member.IsAbstract`. That's probably better than `SymbolKind.NamedType` check I suggested earlier.

**Comment by jkoritzinsky** [comment]
We could have the code-fix not provide a fix for this scenario since there’s no way to provide an implementation.

static abstracts and IDIC don’t mix well. One requires static type system info and the other is exclusively dynamically used.

**Comment by Youssef1313** [comment]
The compiler no longer produces an error for the scenario I mentioned. So no change in the codefix will be needed
